sbi/inference/snpe/snpe_a.py

```diff
@@ -1,36 +1,315 @@
-        summary_writer: Optional[SummaryWriter] = None,
+        summary_writer: Optional[TensorboardSummaryWriter] = None,
         show_progress_bars: bool = True,
-        show_round_summary: bool = False,
+        **unused_args,
     ):
-        """SNPE-A [1]. CURRENTLY NOT IMPLEMENTED.
+        r"""SNPE-A [1].
+
+        https://github.com/mackelab/sbi/blob/main/sbi/inference/snpe/snpe_c.py
+        https://github.com/mackelab/sbi/blob/main/sbi/neural_nets/mdn.py
 
         [1] _Fast epsilon-free Inference of Simulation Models with Bayesian Conditional
             Density Estimation_, Papamakarios et al., NeurIPS 2016,
             https://arxiv.org/abs/1605.06376.
 
+        Args:
+            prior: A probability distribution that expresses prior knowledge about the
+                parameters, e.g. which ranges are meaningful for them. Any
+                object with `.log_prob()`and `.sample()` (for example, a PyTorch
+                distribution) can be used.
+            density_estimator: If it is a string, use a pre-configured network of the
+                provided type (one of nsf, maf, mdn, made). Alternatively, a function
+                that builds a custom neural network can be provided. The function will
+                be called with the first batch of simulations (theta, x), which can
+                thus be used for shape inference and potentially for z-scoring. It
+                needs to return a PyTorch `nn.Module` implementing the density
+                estimator. The density estimator needs to provide the methods
+                `.log_prob` and `.sample()`.
+            num_components:
+                Number of components of the mixture of Gaussians. This number is set to 1 before
+                running Algorithm 1, and then later set to the specified value before running
+                Algorithm 2.
+            num_rounds: Total number of training rounds. For all but the last round, Algorithm 1
+                from [1] is executed. For last round, Algorithm 2 from [1] is executed once.
+                By default, `num_rounds` is set to 1, i.e. only Algorithm 2 is executed once
+                without training the proposal prior using Algorithm 1.
+            device: torch device on which to compute, e.g. gpu, cpu.
+            logging_level: Minimum severity of messages to log. One of the strings
+                INFO, WARNING, DEBUG, ERROR and CRITICAL.
+            summary_writer: A tensorboard `SummaryWriter` to control, among others, log
+                file location (default is `<current working directory>/logs`.)
+            show_progress_bars: Whether to show a progressbar during training.
+            unused_args: Absorbs additional arguments. No entries will be used. If it
+                is not empty, we warn. In future versions, when the new interface of
+                0.14.0 is more mature, we will remove this argument.
+        """
+
```

Comment:
done for the `density_estimator`.

also done for the `prior`, however I reverted my commit of throwing an error since it makes `test_inference_with_user_sbi_problems()` fail even though SNPE-A is actually working with the custom priors that are wrapped to be a `BoxUniform`.